src/GitHub.Api/SimpleApiClient.cs

```diff
@@ -71,10 +71,10 @@ public class SimpleApiClient : ISimpleApiClient
                 }
             }
             // it'll throw if it's private or an enterprise instance requiring authentication
-            catch (ApiException)
+            catch (ApiException apiex)
             {
                 if (!HostAddress.IsGitHubDotComUri(OriginalUrl.ToRepositoryUrl()))
-                    isEnterprise = true;
+                    isEnterprise = apiex.HttpResponse?.Headers.ContainsKey("X-GitHub-Request-Id");
```

Comment:
Ooooh. Could you check GitHub Desktop to see if we're doing the same thing there?